src/test/regress/expected/merge.out

```diff
@@ -1213,9 +1336,1277 @@ SELECT * FROM ft_target;
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$merge_schema.citus_pa_source$$)
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE OR REPLACE FUNCTION pa_compare_tables() RETURNS BOOLEAN AS $$
+DECLARE ret BOOL;
+BEGIN
+SELECT count(1) = 0 INTO ret
+    FROM pg_pa_target
+    FULL OUTER JOIN citus_pa_target
+        USING (tid, balance, val)
+    WHERE pg_pa_target.tid IS NULL
+        OR citus_pa_target.tid IS NULL;
+RETURN ret;
+END
+$$ LANGUAGE PLPGSQL;
+-- try simple MERGE
+BEGIN;
+MERGE INTO pg_pa_target t
+  USING pg_pa_source s
+  ON t.tid = s.sid
+  WHEN MATCHED THEN
+    UPDATE SET balance = balance + delta, val = val || ' updated by merge'
+  WHEN NOT MATCHED THEN
+    INSERT VALUES (sid, delta, 'inserted by merge');
+MERGE INTO citus_pa_target t
+  USING citus_pa_source s
+  ON t.tid = s.sid
+  WHEN MATCHED THEN
+    UPDATE SET balance = balance + delta, val = val || ' updated by merge'
+  WHEN NOT MATCHED THEN
+    INSERT VALUES (sid, delta, 'inserted by merge');
+SELECT pa_compare_tables();
+ pa_compare_tables
+---------------------------------------------------------------------
+ t
+(1 row)
+
+ROLLBACK;
+-- same with a constant qual
+BEGIN;
+MERGE INTO pg_pa_target t
+  USING pg_pa_source s
+  ON t.tid = s.sid AND tid = 1
+  WHEN MATCHED THEN
+    UPDATE SET balance = balance + delta, val = val || ' updated by merge'
+  WHEN NOT MATCHED THEN
+    INSERT VALUES (sid, delta, 'inserted by merge');
+MERGE INTO citus_pa_target t
+  USING citus_pa_source s
+  ON t.tid = s.sid AND tid = 1
+  WHEN MATCHED THEN
+    UPDATE SET balance = balance + delta, val = val || ' updated by merge'
+  WHEN NOT MATCHED THEN
+    INSERT VALUES (sid, delta, 'inserted by merge');
+SELECT pa_compare_tables();
+ pa_compare_tables
+---------------------------------------------------------------------
+ t
+(1 row)
+
+ROLLBACK;
+CREATE TABLE source_json( id   integer, z int, d jsonb);
+CREATE TABLE target_json( id   integer, z int, d jsonb);
+INSERT INTO source_json SELECT i,i FROM generate_series(0,100)i;
+SELECT create_distributed_table('target_json','id'), create_distributed_table('source_json', 'id');
+NOTICE:  Copying data from local table...
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$merge_schema.source_json$$)
+ create_distributed_table | create_distributed_table
+---------------------------------------------------------------------
+                          |
+(1 row)
+
+-- single shard query given source_json is filtered and Postgres is smart to pushdown
+-- filter to the target_json as well
+EXPLAIN MERGE INTO target_json sda
```

Comment:
In general, we try to avoid the EXPLAIN outputs on the workers/tasks, as they are hard to maintain across PG versions.


If we are only interested in the task count, shall we use `coordinator_plan` helper function? https://github.com/citusdata/citus/blob/e93793593565b727c7bfb0531a16e851b4cd8fe3/src/test/regress/expected/pg12.out#L120


(same for the following queries)

Reply:
The ideas was to check the filter on target pushed down
```
->  Seq Scan on target_json_4000088 sda  (cost=0.00..25.00 rows=6 width=10)
                                 Filter: (id = 1)
```
If it's hard to maintain across the versions, it will be problematic with flaky results. I've changed it to coordinator plan.